File: csharp/2011/the-adjustment-bureau-add.cs
using System;
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace MovieDatabase
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Create a MovieRepository instance
            var movies = new MovieRepository();

            // Add "The Adjustment Bureau" to the database
            // This demonstrates how to insert a new item into DynamoDB
            await movies.InsertAsync(
                title: "The Adjustment Bureau",
                year: 2011,
                plot: "The affair between a politician and a contemporary dancer is affected by mysterious forces keeping the lovers apart.",
                rating: 7.1
            );

            // Confirm that the movie was added by retrieving it
            var movie = await movies.SelectAsync(
                title: "The Adjustment Bureau",
                year: 2011
            );

            if (movie != null)
            {
                // The movie was found
                Console.WriteLine($"Movie found: {movie}");
            }
            else
            {
                // The movie was not found
                Console.WriteLine("Movie not found");
            }
        }
    }
}

File: python/2011/the-adjustment-bureau-add.py

from movies import MovieRepository

# Create a new movie repository using the default configuration
movies = MovieRepository()

# Add The Adjustment Bureau to the database
movies.insert(
    title="The Adjustment Bureau", 
    year=2011, 
    plot="The affair between a politician and a contemporary dancer is affected by mysterious forces keeping the lovers apart.", 
    rating=7.1
)

# Confirm that the movie was added
movie = movies.select(
    title="The Adjustment Bureau", 
    year=2011
)
if movie:
    # The movie was found
    print(f"Movie found: {movie}")
else:
    # The movie was not found
    print("Movie not found")
    